### PR TITLE
tracing: reduce the number of branches to test

### PR DESCRIPTION
**Pull request by sitano**
tracing: reduce the number of branches to test when tracing is disabled.

### CODE REVIEW

**Comment by sitano** [comment]
cc @vladzcloudius whom shall I ping for it?

File: transport/server.cc [hunk 1]
```diff
@@ -989,7 +989,7 @@ process_query_internal(service::client_state& client_state, distributed<cql3::qu
     auto& options = *q_state->options;
     auto skip_metadata = options.skip_metadata();
 
-    if (init_trace) {
+    if (init_trace && trace_state) {
```
Comment (vladzcloudius) [inline transport/server.cc:992]: `trace_state` is tested in each of these inline functions. So, I'd expect this optimization not to help much. Do you have benchmarks data to compare?

**Review by vladzcloudius — CHANGES_REQUESTED** [review]

**Comment by vladzcloudius** [comment]
> clang is capable of joining consequent checks into 1 if it can prove it does not escape the flow in the middle even in -O1 https://godbolt.org/z/c34nYz74W

Told'ya ;)

**Comment by slivne** [comment]
CLANG is not necessarily going to stay forever - Avi will consider
switching back once gcc has adequate support.

**Comment by vladzcloudius** [comment]
On 4/13/21 2:40 PM, Shlomi Livne wrote:
> CLANG is not necessarily going to stay forever - Avi will consider
> switching back once gcc has adequate support.

I'd guess that gcc would do the same optimization here.

>
> On Tue, Apr 13, 2021 at 8:43 PM Vladislav Zolotarov <

**Pull request closed without merging**